### PR TITLE
Error reporting via Sentry

### PR DESCRIPTION
**Pull request by adrianmroz**

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -760,3 +760,12 @@ These timezones will appear in the dropdown instead of the default, which are
 
 `['America/Juneau', 'America/Los_Angeles', 'America/Yellowknife', 'America/Phoenix', 'America/Denver', 'America/Mexico_City', 'America/Chicago', 'America/New_York', 'America/Argentina/Buenos_Aires', 'Etc/UTC',
 'Asia/Jerusalem', 'Europe/Paris', 'Asia/Kathmandu', 'Asia/Hong_Kong', 'Asia/Seoul', 'Pacific/Guam']`
+
+### Sentry DSN
+
+Add you Sentry DSN to report errors to Sentry
```
Comment (mkuthan) [inline docs/configuration.md]: Typo: "your" or just "Sentry DSN ...". Link to the Sentry documentation https://docs.sentry.io/clients/java/config/

File: src/client/utils/error-reporter/error-reporter.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  static captureMessage(msg: string, level = Sentry.Severity.Log) {
+    Sentry.captureMessage(msg, level);
+  }
+
+  static init(dsn: string) {
+    Sentry.init({ dsn });
+
+    window.onerror = (message, file, line, column, errorObject) => {
+
+      Sentry.captureException(new Error(message.toString()));
```
Comment (mkuthan) [inline src/client/utils/error-reporter/error-reporter.ts]: code duplication, use ErrorReporter.captureError

File: src/client/utils/error-reporter/error-reporter.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015-2016 Imply Data, Inc.
```
Comment (mkuthan) [inline src/client/utils/error-reporter/error-reporter.ts]: Only allegro, I suppose

File: src/client/utils/error-reporter/error-reporter.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Sentry from "@sentry/browser";
+
+export default class ErrorReporter {
```
Comment (mkuthan) [inline src/client/utils/error-reporter/error-reporter.ts]: I would define interface ErrorReporter with single implementation SentryErrorReporter and get rid of static methods ...

File: src/client/applications/turnilo-application/turnilo-application.tsx [hunk 5]
```diff
@@ -79,10 +83,19 @@ export class TurniloApplication extends React.Component<TurniloApplicationProps,
...
+      showAboutModal: false,
+      error: null
     };
   }
 
+  componentDidCatch(error: Error) {
+    reportError(error);
+    this.setState({
+      viewType: ERROR,
+      error: error.message
```
Comment (mpodlasin) [inline src/client/applications/turnilo-application/turnilo-application.tsx]: So if an error happens, I will get only error message?

How about stack trace of where error happened? I cannot live without it!

Comment (mpodlasin) [reply src/client/applications/turnilo-application/turnilo-application.tsx]: https://reactjs.org/docs/react-component.html#componentdidcatch - there is some stack trace provided by React as second parameter.

Comment (mpodlasin) [reply src/client/applications/turnilo-application/turnilo-application.tsx]: It just seems easier flow to me (just see stack trace and immediately go to line that caused problems vs go to sentry, find logs etc. But I never used sentry. Maybe it's amazing and will make debugging easier. If so, ignore my comments.

**Pull request merged**